services/giphy/giphy.go

```diff
@@ -130,16 +133,16 @@ func (s *Service) searchGiphy(query string) (*result, error) {
 	return &search.Data, nil
 }
 
-func asInt(strInt string) uint {
-	u64, err := strconv.ParseUint(strInt, 10, 32)
+func asInt(strInt string) int {
+	u64, err := strconv.ParseInt(strInt, 10, 32)
```

Comment:
Why did you change this?

Reply:
Because all of `FileInfo`s fields are ints in mautrix (and not uints as in gomatrix).